dubbo-container/dubbo-container-spring/src/main/java/com/alibaba/dubbo/container/spring/SpringContainer.java

```diff
@@ -44,6 +45,7 @@ public void start() {
             configPath = DEFAULT_SPRING_CONFIG;
         }
         context = new ClassPathXmlApplicationContext(configPath.split("[,\\s]+"));
+        context.addApplicationListener(new DubboApplicationListener());

```

Comment:
If you are interested in ContextRefreshedEvent or other start events, you will not receive a notification. Unless specified not to refresh, eg: 
```
context = new ClassPathXmlApplicationContext(configPath.split("[,\\s]+"), false);
context.addApplicationListener(new DubboApplicationListener());
context.refresh();
```

Reply:
@ralf0131
spring-boot defaults to registration hooks. Spring does not register by default.
Registering hooks by default is a good idea.
eg:
```
 if(context instanceof AbstractApplicationContext) {
     AbstractApplicationContext ctx = (AbstractApplicationContext)context;
     ctx.registerShutdownHook();
}
```

Reply:
Yes, we should call refresh(). I will fix it.

Reply:
Yes, we should also call `registerShutdownHook()` to ensure there is at least one shutdown hook registered, because the current implementation assumes that when Dubbo detects it is running under spring, it will automatically remove the previously registered shutdownhook.
Thanks for the reminder.